src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.container.js

```diff
@@ -28,11 +29,12 @@ const useComponentLifecycle = () => {
         }), shallowEqual);
 
     const { scopeType } = useScopeInfo(programId);
-    const { programHasEnrollments, enrollmentsOnProgramContainEnrollmentId } = useEnrollmentInfo(enrollmentId, programId);
+    const { programHasEnrollments, enrollmentsOnProgramContainEnrollmentId, autoEnrollmentId } = useEnrollmentInfo(enrollmentId, programId);
     useEffect(() => {
         const selectedProgramIsTracker = programId && scopeType === scopeTypes.TRACKER_PROGRAM;
-
-        if (selectedProgramIsTracker && programHasEnrollments && enrollmentsOnProgramContainEnrollmentId) {
+        if (enrollmentId === 'AUTO' && autoEnrollmentId) {
+            dispatch(setEnrollmentSelection({ enrollmentId: autoEnrollmentId }));
```

Comment:
Can we, instead of going with the epics, use the hook that is defined here:
https://github.com/dhis2/capture-app/blob/0772559f4fd54f53b44d9fc1478c36e07716b027/src/core_modules/capture-core/components/ScopeSelector/hooks/useSetEnrollmentId.js#L6

We need to make one change though; use history.replace instead of history.push (if not, you will get a hard time going back with the browser back button; you will be forwarded again). So, make it an option to replace instead of pushing.

Reply:
👍  thanks @JoakimSM  I update the changes as suggested